### ISSUE TITLE
PYTHONPATH environment variable not inherited if using non-default python interpreter

### ISSUE THREAD

**Issue by mrclary** [spyder-ide/spyder#8681]
## Description
It seems that the iPython console does not inherit `PYTHONPATH` consistently in Spyder.

### What steps will reproduce the problem?
The two scenarios described below illustrate the issue. In both scenarios Spyder is launched from the commandline within an Anaconda environment dedicated for running Spyder. Also, Spyder's PYTHONPATH Manager and the user's `PYTHONPATH` environment variable (defined in ~/.bash_profile) are different.

#### Scenario 1: iPython inherits `PYTHONPATH` from Spyder's PYTHONPATH Manager
1. From the preferences, select "Use the following Python interpreter:". I chose the one from my base Anaconda environment (not the Spyder launch environment).
2. Open a new iPython console
3. Value of `os.environ['PYTHONPATH']` reflects the contents of Spyder's PYTHONPATH Manager, not the user's `PYTHONPATH` environment variable, though other user environment variables defined in ~/.bash_profile are present.

#### Scenario 2: iPython inherits `PYTHONPATH` from user's `PYTHONPATH` environment variable
1. From the preferences, select "Default (i.e. same as Spyder's)"
2. Open a new iPython console
3. Value of `os.environ['PYTHONPATH']` reflects the user's `PYTHONPATH` environment variable.

### Request
Spyder should consistently apply the `PYTHONPATH`, either from the user's environment variables or from Spyder's PYTHONPATH Manager. I'd prefer that the user be able to control which one is used, otherwise default to the user's environment variable.

## Versions

* Spyder version: 3.3.2 
* Python version: 3.7.0
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Darwin 18.2.0

### Dependencies

```
pyflakes >=0.6.0  :  2.0.0 (OK)
pycodestyle >=2.3 :  2.4.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
sphinx >=0.6.6    :  1.8.1 (OK)
rope >=0.9.4      :  0.11.0 (OK)
jedi >=0.9.0      :  0.13.1 (OK)
psutil >=0.3      :  5.4.8 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
pandas >=0.13.1   :  None (NOK)
numpy >=1.7       :  None (NOK)
sympy >=0.7.3     :  None (NOK)
cython >=0.21     :  None (NOK)
qtconsole >=4.2.0 :  4.4.3 (OK)
IPython >=4.0     :  7.1.1 (OK)
matplotlib >=2.0.0:  None (NOK)
pylint >=0.25     :  2.1.1 (OK)
```

**Comment by ccordoba12** [comment]
Thanks for reporting. We'll fix in our next version (3.3.3).

**Comment by why-not** [comment]
Observing the same behavior in ipython version 7.19.0
